Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM golang:1.18.3 as builder
+
+COPY . /go/src/github.com/kong/go-echo
+
+WORKDIR /go/src/github.com/kong/go-echo
+RUN CGO_ENABLED=0 go build -o /go/bin/tcp-echo .
+
+FROM alpine:3.16.0
+
+RUN apk --no-cache add ca-certificates
+
+COPY --from=builder /go/bin/tcp-echo /tcp-echo
+
+WORKDIR /
+
+ENTRYPOINT ["/tcp-echo"]
```

Comment:
Oh and BTW: you might want to consider using [distroless](https://github.com/GoogleContainerTools/distroless) which is basically a `scratch` image with:

- ca-certificates
- A /etc/passwd entry for a root user
- A /tmp directory
- tzdata

for `static` flavor, or having additionally (for those `cgo` binaries):

- glibc
- libssl
- openssl

for `base` images.

They also claim to be:

> The smallest distroless image, gcr.io/distroless/static-debian11, is around 2 MiB. That's about 50% of the size of alpine (~5 MiB)

Reply:
It's a good idea, I used it instead of Alpine :)

Reply:
@pmalek In the end I moved back from distroless to Alpine as Distroless doesn't have specific versioning, and we'd end up in a situation very similar to the `httpbin` one for what is concerning the base image versioning.

Reply:
Hmm I see. So the only thing we could do is to pin to a specific SHA e.g. from here https://console.cloud.google.com/gcr/images/distroless/GLOBAL/static-debian11

But OTOH distroless is meant to be just an image (almost empty one) with certs, tzdata, tmp dir  and `nonroot` user. So maybe (just maybe :) ) it's not worth paying attention to its versioning like in case of alpine which in my view is a bit different (having its own package manager etc.).

WDYT?